File: compiler/bench/cmp.c
// expected: 750000000
#include <stdio.h>
int main() {
    long sum = 0;
    for (long i = 0; i < 1000000000; i++) {
        if ((i & 3) == 0) sum++;
        if ((i & 4) > 0) sum++;
        if (i < 500000000) sum++;
    }
    printf("%ld\n", sum);
    return 0;
}

File: compiler/bench/cmp.fs
\ expected: 750000000
\ Comparison operators stress - <, >, 0=, 0<, >=

: main
  0 1000000000 0 do
    i 3 and 0= if 1+ then
    i 4 and 0> if 1+ then
    i 500000000 < if 1+ then
  loop . cr ;
